### ISSUE TITLE
Functional Models with multiple outputs and losses are not aligned on `model.fit` history with tf.keras

### ISSUE THREAD

**Issue by sampathweb** [keras-team/keras-core#361]
### Scenario Model has list of outputs and has list of Losses & Metrics:

```
def get_model():
    x = keras.layers.Input(shape=(3,), name="input_a")
    output_a = keras.layers.Dense(1, name="output_a")(x)
    output_b = keras.layers.Dense(1, name="output_b", activation="sigmoid")(x)
    model = keras.models.Model(x, [output_a, output_b])
    return model

model = get_model()

x = np.random.rand(8, 3)
y1 = np.random.rand(8, 1)
y2 = np.random.randint(0, 2, (8, 1))
model.compile(
    optimizer="sgd",
    loss=["mse", "binary_crossentropy"],
    metrics=[["mse"], ["mse", "accuracy"]]
)
hist = model.fit(x, (y1, y2), batch_size=2, epochs=1, verbose=0)
print(hist.history.keys())
```

In `tf.keras` the list of keys in history (correct result) - 
```
dict_keys(['loss', 'output_a_loss', 'output_b_loss', 'output_a_mse', 'output_b_mse', 'output_b_accuracy'])
```

In `keras_core`, the list of keys in history - 
```
dict_keys(['accuracy', 'loss', 'mean_squared_error', 'mean_squared_error_1'])
```

There are three issues - 
1. `mse` is named `mean_squared_error` - This might be intended behavior?
2. `loss` value only has combined loss, and we no longer get individual losses in multi-outputs.
3. `metrics` are not attributed to individual outputs.

We may need to resolve this issue before #352
